src/main/java/htsjdk/samtools/BAMFileReader.java

```diff
@@ -763,7 +801,7 @@ private void assertIntervalsOptimized(final QueryInterval[] intervals) {
     /**
      * Iterate over the SAMRecords defined by the sections of the file described in the ctor argument.
      */
-    private class BAMFileIndexIterator extends BAMFileIterator {
+    public class BAMFileIndexIterator extends BAMFileIterator {
```

Comment:
We were doing a collaborative review on this branch, so we needed our comments to be visible to everyone right away rather than at the end of the review. Sorry for the inconvenience.